test/e2e/framework/fixture.go

```diff
@@ -385,18 +402,66 @@ func (sf SyncerFixture) Start(t *testing.T) *StartedSyncerFixture {
 	downstreamKubeClient, err := kubernetesclientset.NewForConfig(downstreamConfig)
 	require.NoError(t, err)
 
-	if useDeployedSyncer {
-		// Ensure cleanup of pcluster resources
+	artifactDir, err := CreateTempDirForTest(t, "artifacts")
+	if err != nil {
+		t.Errorf("failed to create temp dir for syncer artifacts: %v", err)
+	}
+
+	// collect both in deployed and in-process mode
+	t.Cleanup(func() {
+		ctx, cancelFn := context.WithDeadline(context.Background(), time.Now().Add(time.Second*30))
+		defer cancelFn()
+
+		t.Logf("Collecting imported resource info")
+		upstreamCfg := sf.UpstreamServer.DefaultConfig(t)
+
+		gather := func(client dynamic.Interface, gvr schema.GroupVersionResource) {
+			resourceClient := client.Resource(gvr)
+
+			t.Logf("gathering %q", gvr)
+			list, err := resourceClient.List(ctx, metav1.ListOptions{})
+			require.NoError(t, err, "error listing %q", gvr)
 
+			t.Logf("got %d items", len(list.Items))
+
+			for i := range list.Items {
+				item := list.Items[i]
+				sf.UpstreamServer.Artifact(t, func() (runtime.Object, error) {
+					return &item, nil
+				})
+			}
+		}
+
+		upstreamDynamic, err := dynamic.NewClusterForConfig(upstreamCfg)
+		require.NoError(t, err, "error creating upstream dynamic client")
+
+		downstreamDynamic, err := dynamic.NewForConfig(downstreamConfig)
+		require.NoError(t, err, "error creating downstream dynamic client")
+
+		gather(upstreamDynamic.Cluster(sf.WorkspaceClusterName), apiresourcev1alpha1.SchemeGroupVersion.WithResource("apiresourceimports"))
+		gather(upstreamDynamic.Cluster(sf.WorkspaceClusterName), apiresourcev1alpha1.SchemeGroupVersion.WithResource("negotiatedapiresources"))
+		gather(upstreamDynamic.Cluster(sf.WorkspaceClusterName), corev1.SchemeGroupVersion.WithResource("namespaces"))
+		gather(downstreamDynamic, corev1.SchemeGroupVersion.WithResource("namespaces"))
+		gather(upstreamDynamic.Cluster(sf.WorkspaceClusterName), appsv1.SchemeGroupVersion.WithResource("deployments"))
+		gather(downstreamDynamic, appsv1.SchemeGroupVersion.WithResource("deployments"))
+	})
+
+	if useDeployedSyncer {
 		syncerID := syncerConfig.ID()
 		t.Cleanup(func() {
-			if useDeployedSyncer {
-				t.Logf("Collecting syncer %s logs", syncerID)
-
-				ctx, cancelFn := context.WithDeadline(context.Background(), time.Now().Add(time.Second*30))
-				defer cancelFn()
+			ctx, cancelFn := context.WithDeadline(context.Background(), time.Now().Add(time.Second*30))
+			defer cancelFn()
 
-				pods, err := downstreamKubeClient.CoreV1().Pods(syncerID).List(ctx, metav1.ListOptions{})
+			// collect syncer logs
+			pods, err := downstreamKubeClient.CoreV1().Pods(syncerID).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				t.Errorf("failed to list pods in %s: %v", syncerID, err)
```

Comment:
will `pods` will be `nil` in this case? (safe to access `pods.Items`?)

Reply:
Needs fixing

Reply:
Was a weird merge. Fixed